src/elements/checkbox/checkbox.js

```diff
@@ -0,0 +1,62 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('semantic.ui.elements.checkbox', [])
+
+    .directive('smCheckbox', smCheckbox);
+
+  function smCheckbox($parse) {
+    return {
+      restrict: 'E',
+      transclude: true,
+      replace: true,
+      scope: true,
+      template: '<div class="ui checkbox">' +
+                  '<input type="checkbox">' +
+                  '<label ng-transclude></label>' +
+                '</div>',
+
+      link: function(scope, element, attrs) {
+        var checked = false;
+        var disabled = false;
+        var input = angular.element(element).find('input');
+
+        element.on('click', toggleFn);
+
+        scope.$watch(attrs.ngDisabled, function(val) {
+          disabled = val || false;
+          input.attr('disabled', disabled);
+        });
+
+        scope.$watch(attrs.ngModel, function(val) {
+          checked = val || false;
+          input.attr('checked', checked);
+        });
+
+        if (attrs.toggle !== void 0) {
+          angular.element(element).addClass('toggle');
+        }
+        else if (attrs.slider !== void 0) {
+          angular.element(element).addClass('slider');
+        }
+
+        if (attrs.ariaLabel === void 0) {
+          element.attr('aria-label', element[0].textContent.trim());
+        }
+
+        function toggleFn() {
+
+          if (disabled) { return; }
+
+          checked = !checked;
+          input.attr('checked', checked);
+          $parse(attrs.ngModel).assign(scope, checked);
```

Comment:
ngModel is great for something so simple, I said avoid isolate scopes (and new scopes). We should only be making a new scope if the user explicitly asks for one